x/stakeibc/keeper/msg_server.go

```diff
@@ -277,6 +285,23 @@ func (k msgServer) RegisterHostZone(goCtx context.Context, msg *types.MsgRegiste
 	return &types.MsgRegisterHostZoneResponse{}, nil
 }
 
+func (ms msgServer) UpdateHostZoneParams(goCtx context.Context, msg *types.MsgUpdateHostZoneParams) (*types.MsgUpdateHostZoneParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if ms.authority != msg.Authority {
```

Comment:
Sorry, this might be a silly question, what does this line do? Is `ms.authority` the gov module?

Reply:
not a silly question! authority is the gov module account - so this is how gov transactions are handled in sdk 47+

You can run `submit-proposal` with any arbitrary sdk message, the gov module will route it to the msg_server with the module address as the "signer" and then, if you want to restrict the tx to gov-only, you have to add this clause to confirm that the signer is the gov module address!

Reply:
What prevents users from calling this function (by setting msg.Authority)?

Reply:
they can't set `msg.Authority` cause that would require them signing a tx from the gov module account